tycho-p2-repository-plugin/src/main/java/org/eclipse/tycho/plugins/p2/repository/ModifyRepositoryMetadataMojo.java

```diff
@@ -0,0 +1,159 @@
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Hannes Wellmann - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.tycho.plugins.p2.repository;
+
+import java.io.IOException;
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugin.MojoFailureException;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.codehaus.plexus.logging.Logger;
+import org.eclipse.equinox.p2.core.ProvisionException;
+import org.eclipse.tycho.core.maven.AbstractP2Mojo;
+import org.eclipse.tycho.p2.tools.RepositoryModifier;
+import org.eclipse.tycho.p2.tools.RepositoryModifier.ModifiedRepositoryDescriptor;
+import org.eclipse.tycho.p2tools.copiedfromp2.RepositoryDescriptor;
+
+/**
+ * Modifies metadata of a P2-repository like its name or properties.
+ * <p>
+ * This can be used to customize metadata of a just assembled repository or to modify an already
+ * existing (remote) repository. The modified repository metadata files
+ * ({@code artifact.xml/jar/xml.xz} or {@code content.xml/jar/xml.xz}) are always written to a local
+ * file. For a remote repository a suitable other tool (like SSH) must be used to copy these files
+ * subsequently back to the remote server.
+ * </p>
+ */
+@Mojo(name = "modify-metadata", threadSafe = true)
+public class ModifyRepositoryMetadataMojo extends AbstractP2Mojo {
+
+    /**
+     * Location of the <em>source</em> repository to modify.<br>
+     * May be the {@code Path} of a local file or {@code URL}/{@code URI} to a remote
+     * repository.<br>
+     * If this is set to a remote URL the {@link #outputLocation} must be set to a local path as
+     * remote locations cannot be written.
+     */
+    @Parameter(property = "p2.repository.location")
+    private URI repositoryLocation;
+
+    /**
+     * Optional output location where the modified metadata files ({@code artifact.xml/jar/xml.xz}
+     * or {@code content.xml/jar/xml.xz}) are written to.<br>
+     * If not specified or equals to {@link #repositoryLocation} the source repository is modified
+     * in-place.
+     */
+    @Parameter(property = "p2.repository.output")
+    private Path outputLocation;
+
+    /**
+     * The kind of modified repository: Either {@code artifact} or {@code metadata}<br>
+     * If {@code artifact} is specified, only the {@code artifact.xml/jar/xml.xz} files are
+     * modified, if {@code metadata} only {@code content.xml/jar/xml.xz}. If blank (the default)
+     * both are modified.
+     */
+    @Parameter(property = "p2.repository.kind", defaultValue = "")
+    private String repositoryKind;
+
+    /** Optional repository name to set. An existing name is overwritten. */
+    @Parameter
```

Comment:
Do we want a user property here?
Same for the below parameters.

Reply:
I can add them, but I assume that one cannot configure the mojo via a property for a `Map` parameter, can one?

Reply:
It works for lists at least:
https://maven.apache.org/guides/mini/guide-configuring-plugins.html#Mapping_Collections_and_Arrays

> Since Maven 3.3.9 ([MNG-5440](https://issues.apache.org/jira/browse/MNG-5440)), you can list individual items alternatively as comma-separated list in the XML value of animals directly. This approach is also used if configuring collection/array parameters via command line

So for maps one would need to add support, or we allow another parameter as a list of `key=value` pairs.

Reply:
A little bit further down this guides it says:
> In contrast to value objects and collections/arrays there is no string coercion defined for maps, i.e. you cannot give parameters of that type via CLI argument.

> or we allow another parameter as a list of key=value pairs.

That would be possible, but I suggest to save this part for now as I assume that one will provide specific executions for intended tasks in the `pluginManagement` section anyways in a pom (and we need a pom at least to define the tycho-version or we have to hard-code it into a using pipeline).